mk/spksrc.common.mk

```diff
@@ -57,19 +57,6 @@ SUPPORTED_ARCHS = $(sort $(filter-out $(ARCHS_DUPES), $(AVAILABLE_TOOLCHAINS)))
 #         all archs except generic archs
 LEGACY_ARCHS = $(sort $(filter-out $(addsuffix %,$(GENERIC_ARCHS)), $(AVAILABLE_TOOLCHAINS)))
 
-
-# Relocate to set conditionally according to existing parallel options in caller
-ifneq ($(PARALLEL_MAKE),)
-ifeq ($(PARALLEL_MAKE),max)
-NCPUS = $(shell grep -c ^processor /proc/cpuinfo)
-else
-NCPUS = $(PARALLEL_MAKE)
-endif
```

Comment:
If we revert this bit (or do something similar) than we can use $NCPUS to get the correct number for -j. Saves an if statement for checking max every time. Looking at the code the cmake part doesn't handle PMAKE=4 only max or none at the moment.

Reply:
Sure thing.  Note that I was hoping to complete this asap as I will be offline for another week or so.  I'll finish things up upon my return.

Reply:
done, it actually simplify the code overall and allowed to fix cmake as well.